tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+     * @param methodArgs the arguments in the method under analysis
+     * @param expression a Java expression
+     * @return the type of the expression
+     * @throws ResolvedTypeNotFound if {@code jpClass} is not a class/interface
+     * or if {@code jpCallable} is a constructor
+     * @throws NoSuchElementException if an error occurs while parsing the
+     * expression
+     */
+    public static ResolvedType getResolvedTypeOfExpression(
+            TypeDeclaration<?> jpClass,
+            CallableDeclaration<?> jpCallable,
+            List<Triplet<String, String, String>> methodArgs,
+            String expression
+    ) throws ResolvedTypeNotFound {
+        if (jpClass instanceof ClassOrInterfaceDeclaration) {
+            addSyntheticMethodWithExpression(
+                    jpClass,
+                    jpCallable,
+                    methodArgs,
+                    expression
+            );
+            // find expression in synthetic method and resolve return type
+            return jpClass.getMethodsByName(SYNTHETIC_METHOD_NAME).get(0)
+                    .getBody().orElseThrow()
+                    .getStatements().getLast().orElseThrow()
+                    .asExpressionStmt().getExpression()
+                    .asVariableDeclarationExpr().getVariables().get(0)
+                    .getInitializer().orElseThrow()
+                    .calculateResolvedType();
+        }
+        throw new ResolvedTypeNotFound(String.format(
+                "ResolvedType of expression %s of class %s and method %s not found.",
+                expression,
+                jpClass.getNameAsString(),
+                jpCallable.getNameAsString()
+        ));
+    }
+
+    /**
+     * Given a syntactically valid Java expression, evaluates its return type, including package and class.
+     * If the expression is not syntactically valid, the method will throw an exception.
+     * @param expression the expression to evaluate, e.g., "methodResultID.negate().value(null).getField()".
+     *                   Must conform to TrattoGrammar.
+     * @param oracleDatapoint OracleDatapoint containing additional necessary information for resolving types,
+     *                        e.g., class and method names and sources. NOTE: To handle Java expressions
+     *                        containing the token "jdVar", the oracleDatapoint must have its "oracle" field
+     *                        populated, even if it's a partial oracle (e.g., it's the oracle being currently
+     *                        generated). Then, the last occurring jdVar clause is looked for and its type is
+     *                        resolved, which will be used to resolve the type of the expression.
+     * @return pair of strings, where the first string is the package and the second string is the class
+     */
+    public static Pair<String, String> getReturnTypeOfExpression(String expression, OracleDatapoint oracleDatapoint) {
```

Comment:
@AML14

Reply:
There are 67 usages of this method in the project and it doesn't seem that we needed the ResolvedType for anything, only the package and/or the class. This change would also make 67 code lines longer than needed. Lastly, since ResolvedType is an interface, I'm not sure how to create an instance of it that represents `null`, as we need to do right at the beginning of the method:

```java
public static Pair<String, String> getReturnTypeOfExpression(String expression, OracleDatapoint oracleDatapoint) {
        // Handle null
        if ("null".equals(expression)) {
            return JavaTypes.NULL;
        }
```